plugins/engine-configuration/impl/src/test/java/org/pentaho/di/engine/configuration/impl/RunConfigurationProviderFactoryManagerImplTest.java

```diff
@@ -0,0 +1,66 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.engine.configuration.impl;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.pentaho.di.core.service.PluginServiceLoader;
+import org.pentaho.di.engine.configuration.api.CheckedMetaStoreSupplier;
+import org.pentaho.di.engine.configuration.api.RunConfigurationProvider;
+import org.pentaho.di.engine.configuration.api.RunConfigurationProviderFactory;
+import org.pentaho.di.engine.configuration.impl.pentaho.DefaultRunConfigurationProviderFactory;
+import org.pentaho.metastore.stores.memory.MemoryMetaStore;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class RunConfigurationProviderFactoryManagerImplTest {
+
+  @Test
+  public void testDefaultProvidedMetastore() {
+    MemoryMetaStore memoryMetaStore = new MemoryMetaStore();
+    CheckedMetaStoreSupplier metastoreSupplier = () -> memoryMetaStore;
+    List<RunConfigurationProvider> providers =
+      new RunConfigurationProviderFactoryManagerImpl().generateProviders( metastoreSupplier );
+
+    Assert.assertEquals( 1, providers.size() );
```

Comment:
if you are just trying to test the logic for `RunConfigurationProviderFactoryManagerImpl#generateProviders(checkedMetaStoreSupplier )` then you can create a getter/setter function for factory variable - https://github.com/pentaho/pentaho-kettle/pull/9966/files#r2056832326

and test that generic FactoryManagers work.

For unit tests, the goal is to test only the class under Test, RunConfigurationProviderFactoryManagerImpl.java and not neccesarily DefaultRunConfigurationProviderFactory#getProvider returns a correct result

Reply:
ok, ty, I wanted to get into the ServiceProvider piece of this, seems I over-complicated!